lib/couchbase/search_options.rb

```diff
@@ -14,17 +14,73 @@
 
 module Couchbase
   class Cluster
-    class SearchQuery
-      # Prepare {MatchQuery} body
+    # @api volatile
+    class SearchRequest
+      # Creates a search request, used to perform operations against the Full Text Search (FTS) Couchbase service.
       #
-      # @param [String] match
-      # @yieldparam [MatchQuery] query
+      # @overload self.create(search_query)
+      #   Will run an FTS +SearchQuery+
+      #   @param [SearchQuery] search_query
       #
-      # @return [MatchQuery]
-      def self.match(match, &block)
-        MatchQuery.new(match, &block)
+      # @overload self.create(vector_search)
+      #   Will run a +VectorSearch+
+      #   @param [VectorSearch] vector_search
+      def self.create(search)
```

Comment:
Is it a good idea to have the `.create` function for creating `SearchRequest`s instead of a constructor? The RFC has it but it's leaving it open to what's idiomatic for each SDK. For `VectorSearch` and `VectorQuery` I went with just having the constructor which felt more consistent with our existing search APIs. Should I have added `.create` functions for them as well?

Reply:
This method looks like a constructor and from the user perspective it is more natural to call `SearchRequest.new`. Lets rename it.

Reply:
Done